src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+                    break;
+                }
+            }
+            if (func_800A56C8(&this->skelanime, 10.0f)) {
+                BossGoma_PlayEffectsAndSfx(this, globalCtx, 3, 5);
+                func_80033E88(&this->actor, globalCtx, 5, 15);
+            }
+            if (func_800A56C8(&this->skelanime, SkelAnime_GetFrameCount(&D_06004318.genericHeader))) {
+                this->actionState = 1;
+                SkelAnime_ChangeAnim(&this->skelanime, &D_06004A20, 1.0f, 0.0f,
+                                     SkelAnime_GetFrameCount(&D_06004A20.genericHeader), 0, -1.0f);
+                // timer setup to 0 (or 10 when hit(s?), see above)
+                if (this->framesUntilNextAction == 0) {
+                    this->timer = (s16)(Math_Rand_ZeroOne() * 30.0f) + 30;
+                }
+            }
+            break;
+        case 1:
+            if (func_800A56C8(&this->skelanime, 3.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_UNARI2);
+            }
+            // timer set to 30-60 in previous state, or kept previous value...
+            // this is probably meant to add delay if gohma did hit the player
+            if (this->timer == 0) {
+                this->actionState = 2;
+                SkelAnime_ChangeAnim(&this->skelanime, &D_06005330, 1.0f, 0.0f,
+                                     SkelAnime_GetFrameCount(&D_06005330.genericHeader), 2, -5.0f);
+            }
+            break;
+        case 2:
+            if (func_800A56C8(&this->skelanime, SkelAnime_GetFrameCount(&D_06005330.genericHeader))) {
+                BossGoma_SetupFloorIdle(this);
+            }
+            break;
+    }
+
+    this->eyeState = EYESTATE_IRIS_FOLLOW_NO_IFRAMES;
+    this->visualState = VISUALSTATE_RED;
+}
+
+void BossGoma_FloorDamaged(BossGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    if (func_800A56C8(&this->skelanime, SkelAnime_GetFrameCount(&D_06006E60.genericHeader))) {
+        BossGoma_SetupFloorStunned(this);
+        this->patienceTimer = 0;
+    }
+    this->eyeState = EYESTATE_IRIS_NO_FOLLOW_NO_IFRAMES;
+    Math_SmoothScaleMaxF(&this->eyeIrisScaleX, 0.4f, 0.5f, 0.2f);
+    this->visualState = VISUALSTATE_HIT;
+}
+
+void BossGoma_FloorLandStruckDown(BossGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    if (func_800A56C8(&this->skelanime, this->currentAnimFrameCount)) {
+        BossGoma_SetupFloorStunned(this);
+        this->sfxFaintTimer = 92;
+        this->patienceTimer = 0;
+        this->framesUntilNextAction = 150;
+    }
+    func_80033260(globalCtx, &this->actor, &this->actor.posRot.pos, 55.0f, 4, 8.0f, 500, 10, 1);
+}
+
+void BossGoma_FloorLand(BossGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    if (func_800A56C8(&this->skelanime, this->currentAnimFrameCount)) {
+        BossGoma_SetupFloorIdle(this);
+        this->patienceTimer = 200;
```

Comment:
im ont really going to comment much on names cause we arent gonna focus on that too much, but `waitTimer` sounds better than patience to me

Reply:
I named it `patienceTimer` because when it reaches zero Gohma stops moving towards the player and instead runs away, eventually climbing a wall.

Gohma has 7-9 timers depending on your definition of a timer, so I also avoided using names as generic as `waitTimer` (I really dislike using "Wait" everywhere lol). I also had to use one generic `timer` name because it is used for different things so I don't feel like another generic name can be used without confusion.

Reply:
i see. i think theres probably a better word than patience, but not gonna worry too much about it now